pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,7 +618,16 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
+
+            for self_data in self._obs_data():
+                is_equal = False
+                for other_data in other._obs_data():
+                    if np.isclose(self_data[0], other_data[0]):
+                        is_equal = True
+                        Continue
+                if not is_equal:
+                    return False
+            return True
 
         if isinstance(other, (Tensor, Observable)):
             self.simplify()
```

Comment:
Your fix will work when comparing Hamiltonians with other Hamiltonians. I don't know how I forgot this in the last PR, but the changes do not help if Hamiltonians are compared with Tensors or other Observables.

Reply:
Yep, exactly the same. Since the implementation for the comparison is the same either way, you can also just merge them together, though not required.

Reply:
Hey, @mudit2812 I can't replicate the float point error whenever the Hamiltonian data is equal to 1.0, so I have to ask if this change is required.